src/emulator/events/types.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * The types in this file are stolen from the firebase-functions SDK. They are not subject
+ * to change because they are TS interpretations of protobuf wire formats already used in
+ * productiton services.
+ */
+
+/**
+ * Wire formal for v1beta1 EventFlow.
+ * Used by Firestore and some other services.
+ */
+export interface LegacyEvent {
```

Comment:
This type can be imported from [here](https://github.com/firebase/firebase-functions/blob/68a669183f61002c269781b3957bf9b74f10bd87/src/cloud-functions.ts#L34) as long as we reference `firebase-functions` as type only (no code calls) and have it as a dev dep which I think we already do.

Reply:
They're marked as `@internal` though does that matter?

Reply:
```
import { Event, LegacyEvent } from "firebase-functions/lib/cloud-functions" 
```

That doesn't work for me, am I doing it wrong?

Reply:
Yeah they're @internal so they are not exported :-(